src/evict.c

```diff
@@ -712,11 +721,16 @@ int performEvictions(void) {
                 }
             }
         } else {
-            goto cant_free; /* nothing to free... */
+            break;
         }
     }
-    /* at this point, the memory is OK, or we have reached the time limit */
-    result = (isEvictionProcRunning) ? EVICT_RUNNING : EVICT_OK;
+
+    if (mem_freed >= (long long)(mem_used - server.key_eviction_memory)) {
+        /* at this point, the memory is OK, or we have reached the time limit */
+        result = (isEvictionProcRunning) ? EVICT_RUNNING : EVICT_OK;
+    } else {
+        goto cant_free;
+    }
 
 cant_free:
```

Comment:
can we remove this `cant_free` goto label?  Also I think we should only reach out to lazyfree on the "hard" eviction path.

Reply:
Now. it looks like work as I remove the label cant_free